src/mist/io/methods.py

```diff
@@ -4328,6 +4344,101 @@ def get_deploy_collectd_command_coreos(uuid, password, monitor, port=25826):
     return "sudo docker run -d -v /sys/fs/cgroup:/sys/fs/cgroup -e COLLECTD_USERNAME=%s -e COLLECTD_PASSWORD=%s -e MONITOR_SERVER=%s -e COLLECTD_PORT=%s mist/collectd" % (uuid, password, monitor, port)
 
 
+def machine_cost_calculator(m):
+    """
+    Calculates and returns the cost for a VM.
+    Highly provider specific, since there is not a
+    straightforward way to get this info
+
+    Supported providers:
+        Packet.net, DigitalOcean, SoftLayer, AWS, Rackspace, Linode, Vultr
+    TODO: GCE, Azure, NephoScale,
+    HostVirtual
+    """
+    cost = {'indicative_cost_per_hour': 0, 'indicative_cost_per_month': 0}
+    if m.driver.type in [Provider.LINODE, Provider.PACKET, Provider.GCE]:
+        sizes = CloudSize.objects.filter(cloud_provider=m.driver.type)
+    if m.driver.type in [Provider.RACKSPACE, Provider.RACKSPACE_FIRST_GEN]:
+        sizes = CloudSize.objects.filter(cloud_provider=m.driver.type, cloud_region=m.driver.region)
+    now = datetime.now()
+    month_days = calendar.monthrange(now.year, now.month)[1]
+    if m.driver.type in config.EC2_PROVIDERS:
+        # Need to get image in order to specify the OS type
+        # out of the image id
+        instance_image = m.extra.get('image_id')
+        try:
+            os_type = CloudImage.objects.get(cloud_provider=m.driver.type, image_id=instance_image).os_type
+        except:
+            os_type = 'linux'
+        sizes = m.driver.list_sizes()
+        size = m.extra.get('instance_type')
+        for node_size in sizes:
+            if node_size.id == size:
+                plan_price = node_size.price.get(os_type)
+                if not plan_price:
+                    # use the default which is linux
+                    plan_price = node_size.price.get('linux')
+                plan_price = float(plan_price.replace('/hour','').replace('$', ''))
+                # just need the float value
+                cost['indicative_cost_per_hour'] = plan_price
+                cost['indicative_cost_per_month'] = float(plan_price) * 24 * month_days
```

Comment:
Is this necessary? Usually when calculating cost per month, we use 30 days so as to be consistent, even if it's a bit off. Calculating it like this will show a different cost per month, each month.